packages/plugin-ext/src/main/browser/window-state-main.ts

```diff
@@ -53,13 +58,19 @@ export class WindowStateMain implements WindowMain, Disposable {
 
     async $openUri(uriComponent: UriComponents): Promise<boolean> {
         const uri = URI.revive(uriComponent);
-        const url = encodeURI(uri.toString(true));
+        const url = new CoreURI(encodeURI(uri.toString(true)));
         try {
-            this.windowService.openNewWindow(url, { external: true });
+            await open(this.openerService, url);
             return true;
         } catch (e) {
             return false;
         }
     }
 
+    async $asExternalUri(uriComponents: UriComponents): Promise<URI> {
+        const uri = URI.revive(uriComponents);
+        const resolved = await this.externalUriService.resolve(new CoreURI(uri));
+        return resolved['codeUri'];
```

Comment:
I think we should either expose the VS codeUri or do a proper conversion (i.e. through toString()).